server/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -71,27 +89,36 @@ public final Tag render(ApplicantQuestionRendererParams params) {
             String.format("Unhandled error display mode: %s", params.errorDisplayMode()));
     }
 
+    boolean hasQuestionErrors = false;
     ImmutableSet<ValidationErrorMessage> questionErrors =
         validationErrors.getOrDefault(question.getContextualizedPath(), ImmutableSet.of());
     if (!questionErrors.isEmpty()) {
       // Question error text
+      hasQuestionErrors = true;
       questionTextDiv.with(
           BaseHtmlView.fieldErrors(
-              messages, questionErrors, ReferenceClasses.APPLICANT_QUESTION_ERRORS));
+                  messages, questionErrors, ReferenceClasses.APPLICANT_QUESTION_ERRORS)
+              .withId(validationErrorId));
+      // Insert error message to be read first.
+      ariaDescribedByIds.add(0, validationErrorId);
     }
 
     if (question.isRequiredButWasSkippedInCurrentProgram()) {
+      hasQuestionErrors = true;
       String requiredQuestionMessage = messages.at(MessageKey.VALIDATION_REQUIRED.getKeyName());
       questionTextDiv.with(
           div()
+              .withId(requiredErrorId)
               .withClasses(Styles.P_1, Styles.TEXT_RED_600)
               .withText("*" + requiredQuestionMessage));
+      // Insert error message to be read first.
+      ariaDescribedByIds.add(0, requiredErrorId);
     }
 
     return div()
-        .withId(question.getContextualizedPath().toString())
+        .withId(id)
         .withClasses(Styles.MX_AUTO, Styles.MB_8, getReferenceClass(), getRequiredClass())
         .with(questionTextDiv)
-        .with(renderTag(params, validationErrors));
+        .with(renderTag(params, validationErrors, ariaDescribedByIds, hasQuestionErrors));
```

Comment:
suggestion (blocking): it's generally a good practice to avoid passing mutable collections around when avoidable. in this class you could make `ariaDescribedByIds` an `ImmutableList.Builder<String>` and change this line to
```suggestion
        .with(renderTag(params, validationErrors, ariaDescribedByIds.build(), hasQuestionErrors));
```